File: leetcode/01-Two-Sum/python/main.py
class Solution:
  def twoSums(self, nums, target):
    result = []
    map = {}

    for index in range(len(nums)):
      num = nums[index]
      diff = target - num
      if num in map:
        result.append(index)
        result.append(map[num])

      map[diff] = index

    result.sort()
    return result

File: leetcode/01-Two-Sum/go/main.go
package main

func twoSum(nums []int, target int) []int {
	var result []int
	cache := make(map[int]int)

	for i := 0; i < len(nums); i++ {
		_, isCached := cache[nums[i]]
		if isCached {
			result = append(result, cache[nums[i]])
			result = append(result, i)
			break
		}

		diff := target - nums[i]
		cache[diff] = i
	}
	return result
}
